bluesky/examples.py

```diff
@@ -342,32 +410,30 @@ def describe_collect(self):
         return {'stream_name': dd}
 
     def complete(self):
-        self.success = True
-        self.ready = False
-        return self
+        return self._completion_status
 
-    def kickoff(self, start, stop, steps):
-        self._steps = np.linspace(start, stop, steps)
+    def kickoff(self):
         self._data = deque()
 
-        # setup the status object (self) that will be returned by
+        # Setup a status object that will be returned by
         # self.complete(). Separately, make dummy status object
         # that is immediately done, and return that, indicated that
         # the 'kickoff' step is done.
         self._future = self.loop.run_in_executor(None, self._scan)
-        self._future.add_done_callback(lambda x: self._finish())
+        st = SimpleStatus()
+        self._completion_status = st
```

Comment:
This should have some checking that it is not already running.  And probably locking.